### PR TITLE
Fix junk left in pPipelines when pipeline creation is rejected by validation rather than ICD.

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

**Comment by chrisforbes** [comment]
@yavn this addresses the problem introduced by #1191.

**Comment by yavn** [comment]
Thanks for following up.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -6907,32 +6910,35 @@ CreateComputePipelines(VkDevice device, VkPipelineCache pipelineCache, uint32_t
         pPipeState[i] = new PIPELINE_STATE;
         pPipeState[i]->initComputePipeline(&pCreateInfos[i]);
         pPipeState[i]->pipeline_layout = *getPipelineLayout(dev_data, pCreateInfos[i].layout);
-        // memcpy(&pPipeState[i]->computePipelineCI, (const void *)&pCreateInfos[i], sizeof(VkComputePipelineCreateInfo));
 
         // TODO: Add Compute Pipeline Verification
-        skip_call |= !validate_compute_pipeline(dev_data->report_data, pPipeState[i], &dev_data->enabled_features,
+        skip |= !validate_compute_pipeline(dev_data->report_data, pPipeState[i], &dev_data->enabled_features,
                                                 dev_data->shaderModuleMap);
```
Comment (tobine) [inline layers/core_validation.cpp]: reduce indent on this line to re-align with line above

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -6868,31 +6867,35 @@ CreateGraphicsPipelines(VkDevice device, VkPipelineCache pipelineCache, uint32_t
...
+
+    lock.unlock();
+    auto result = dev_data->dispatch_table.CreateGraphicsPipelines(device, pipelineCache, count, pCreateInfos, pAllocator, pPipelines);
+    lock.lock();
+    for (i = 0; i < count; i++) {
+        if (pPipelines[i] == VK_NULL_HANDLE) {
             delete pipe_state[i];
         }
-        lock.unlock();
+        else {
```
Comment (mark-lunarg) [inline layers/core_validation.cpp]: Looks like this probably needs clang-formatting

**Comment by karl-lunarg** [comment]
```
./run_wrap_objects_tests.sh 
Insertion test PASSED
Front insertion test PASSED
Back insertion test PASSED
Middle insertion test PASSED
107c107
< ObjectTracker(ERROR): object: 0x9 type: 20 location: 347 msgCode: 3: OBJ ERROR : For device 0x4c7b560, Descriptor Set Layout object 0x9 has not been destroyed.
---
> ObjectTracker(ERROR): object: 0x9 type: 20 location: 347 msgCode: 3: OBJ ERROR : For device 0x46cb710, Descriptor Set Layout object 0x9 has not been destroyed.
Wrap-objects layer validation tests FAILED - wrap-objects altered the results of the layer validation tests

```
@chrisforbes 
With these changes there is still this error.   Can the fixes in this PR be expanded to cover this last failure?  If not, what is the best way to fix this remaining error?

This test has been failing for a week now, making all of our CI testing marked as failed.  It would be good to resolve these problems soon.

**Comment by chrisforbes** [comment]
@karl-lunarg I'm not sure that's the same issue. I'm happy to look into it, though.

**Comment by karl-lunarg** [comment]
> @karl-lunarg I'm not sure that's the same issue. I'm happy to look into it, though.

Sure.  If you don't get to it soon, please create a new GitHub issue and assign to me, or just let me know directly.

**Comment by karl-lunarg** [comment]
I found an obvious object leak in the AllocDescriptorFromEmptyPool test, which I'm fixing in master directly.  This fixes the problem in the wrap objects test.

**Pull request closed without merging**